relay-sampling/src/lib.rs

```diff
@@ -84,1522 +84,39 @@ pub use dsc::DynamicSamplingContext;
-        let result = match_against_rules(
-            &config,
-            &event,
-            &dsc,
-            Utc.with_ymd_and_hms(1970, 10, 11, 0, 0, 0).unwrap(),
-        );
-        assert_trace_match!(result, 0.03, dsc, 1, 3);
-
-        // early return second rule
-        let event = mocked_event(EventType::Transaction, "transaction", "1.1.1", "testing");
-        let dsc = mocked_dynamic_sampling_context(
-            "root_transaction",
-            "1.1.1",
-            "prod",
-            "vip",
-            "user-id",
-            None,
-        );
-        // We will use a time in the middle of 10th and 11th.
-        let result = match_against_rules(
-            &config,
-            &event,
-            &dsc,
-            Utc.with_ymd_and_hms(1970, 10, 11, 0, 0, 0).unwrap(),
-        );
-        assert!(matches!(result, Some(SamplingMatch { .. })));
-        if let Some(spec) = result {
-            assert!(
-                (spec.sample_rate - 0.45).abs() < f64::EPSILON, // 0.45
-                "did not use the sample rate of the second rule"
-            )
-        }
-
-        // early return third rule
-        let event = mocked_event(EventType::Transaction, "transaction", "1.1.1", "testing");
-        let dsc = mocked_dynamic_sampling_context(
-            "root_transaction",
-            "1.1.1",
-            "testing",
-            "vip",
-            "user-id",
-            None,
-        );
-        // We will use a time in the middle of 10th and 11th.
-        let result = match_against_rules(
-            &config,
-            &event,
-            &dsc,
-            Utc.with_ymd_and_hms(1970, 10, 11, 0, 0, 0).unwrap(),
-        );
-        assert_trace_match!(result, 0.02, dsc, 3);
-    }
-
-    #[test]
-    /// test that the correct match is performed when replay id is present in the dsc.
-    fn test_sampling_match_with_trace_replay_id() {
-        let event = mocked_event(EventType::Transaction, "healthcheck", "1.1.1", "testing");
-        let dsc = mocked_dynamic_sampling_context(
-            "root_transaction",
-            "1.1.1",
-            "prod",
-            "vip",
-            "user-id",
-            Some(Uuid::new_v4()),
-        );
-
-        let result = match_against_rules(
-            &mocked_sampling_config_with_rules(vec![SamplingRule {
-                condition: and(vec![not(eq_null("trace.replay_id"))]),
-                sampling_value: SamplingValue::SampleRate { value: 1.0 },
-                ty: RuleType::Trace,
-                id: RuleId(1),
-                time_range: Default::default(),
-                decaying_fn: Default::default(),
-            }]),
-            &event,
-            &dsc,
-            Utc::now(),
-        );
-        assert_trace_match!(result, 1.0, dsc, 1)
-    }
-
-    #[test]
-    /// Tests that no match is done when there are no matching rules.
-    fn test_get_sampling_match_result_with_no_match() {
```

Comment:
removed because its already tested in     `fn test_get_sampling_match_result_with_no_match()`